llvm/test/CodeGen/AVR/addr-of-ret-addr.ll

```diff
@@ -0,0 +1,21 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 2
+; RUN: llc < %s -mtriple=avr | FileCheck %s
+
+declare ptr @llvm.addressofreturnaddress()
+
+define ptr @a() {
+; CHECK-LABEL: a:
+; CHECK:       ; %bb.0:
+; CHECK-NEXT:    push r28
+; CHECK-NEXT:    push r29
+; CHECK-NEXT:    in r28, 61
+; CHECK-NEXT:    in r29, 62
+; CHECK-NEXT:    mov r24, r28
+; CHECK-NEXT:    mov r25, r29
+; CHECK-NEXT:    adiw r24, 3
```

Comment:
The behavior is the same for all `-mcpu` values. There already is `expand<AVR::SPREAD>` which outside of the PR doesn't have a check for some kind of "SP available". This PR does expose it more, so I guess I can include this check. Do I have to check datasheets for each of the supported targets if it has SPH SPL? Or just each AVR family?

Reply:
Seems like this affects old chips specifically if they had 256B or less. Which we have no way of checking. What GCC seems to do is let the user set `-mtiny-stack` and also maybe have per-architecture defaults

Reply:
GCC accepts two options: SP as implemented here, or short SP (only SPL). Looks like it can be degraded to short SP with `-mtiny-stack`. Full list of AVRs with short SPs as listed by GCC:
```
at90s2313
at90s2323
at90s2333
at90s2343
attiny22
attiny26
at90s4433
attiny13
attiny13a
attiny2313
attiny2313a
attiny24
attiny24a
attiny25
attiny261
attiny261a
```

Reply:
Apparently we have an equivalent small stack subtarget feature, so I used it as a condition in SPREAD and SPWRITE lowering